diff --git a/entitygroupfield.libraries.yml b/entitygroupfield.libraries.yml
index a992a26..659e891 100644
--- a/entitygroupfield.libraries.yml
+++ b/entitygroupfield.libraries.yml
@@ -3,7 +3,6 @@ entitygroupfield.admin:
     - core/jquery
     - core/drupal
     - core/drupalSettings
-    - core/once
     - core/drupal.ajax
     - core/drupal.dropbutton
   css:
diff --git a/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php b/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
index 2e3f207..5729e08 100644
--- a/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
+++ b/src/Plugin/Field/FieldWidget/EntityGroupFieldWidgetBase.php
@@ -389,13 +389,6 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
               '#gcontent_show_warning' => TRUE,
             ];
           }
-
-          $info['remove_button_info'] = [
-            '#type' => 'container',
-            '#markup' => $this->t('You are not allowed to remove this item.'),
-            '#attributes' => ['class' => ['messages', 'messages--warning']],
-            '#access' => !$can_delete,
-          ];
         }
         elseif ($item_mode == 'closed') {
           $links['edit_button'] = [
@@ -426,27 +419,6 @@ abstract class EntityGroupFieldWidgetBase extends WidgetBase implements Containe
               '#attributes' => ['class' => ['messages', 'messages--warning']],
             ];
           }
-
-          $info['edit_button_info'] = [
-            '#type' => 'container',
-            '#markup' => $this->t('You are not allowed to edit this item.'),
-            '#attributes' => ['class' => ['messages', 'messages--warning']],
-            '#access' => !$can_edit && $can_delete,
-          ];
-
-          $info['remove_button_info'] = [
-            '#type' => 'container',
-            '#markup' => $this->t('You are not allowed to remove this item.'),
-            '#attributes' => ['class' => ['messages', 'messages--warning']],
-            '#access' => !$can_delete && $can_edit,
-          ];
-
-          $info['edit_remove_button_info'] = [
-            '#type' => 'container',
-            '#markup' => $this->t('You are not allowed to edit or remove this item.'),
-            '#attributes' => ['class' => ['messages', 'messages--warning']],
-            '#access' => !$can_edit && !$can_delete,
-          ];
         }
         elseif ($item_mode == 'remove') {
 
