diff --git a/core/modules/toolbar/tests/src/Functional/ToolbarAdminMenuTest.php b/core/modules/toolbar/tests/src/Functional/ToolbarAdminMenuTest.php
index 9585c86..dbb368a 100644
--- a/core/modules/toolbar/tests/src/Functional/ToolbarAdminMenuTest.php
+++ b/core/modules/toolbar/tests/src/Functional/ToolbarAdminMenuTest.php
@@ -395,6 +395,33 @@ public function testExternalLink() {
   }
 
   /**
+   * Tests the Manage tray in the Toolbar for various authenticated users.
+   *
+   * The authorized user should not have a Manage tab simply with the 'access
+   * toolbar' permission. They need 'access administration pages' or another
+   * common administrative permission for that.
+   */
+  public function testEmptyMenuTray() {
+    // Log out the admin user because we're testing restricted access.
+    $this->drupalLogout();
+    $this->drupalLogin($this->drupalCreateUser(['access toolbar']));
+    $this->assertResponse(200);
+    // @todo The toolbar div itself still has the id "toolbar-administration".
+    // @see https://www.drupal.org/project/drupal/issues/1044090
+    $this->assertSession()->elementExists('css', 'div[id=toolbar-administration]');
+    $this->assertSession()->elementNotExists('css', 'a[id=toolbar-item-administration]');
+
+    // Try again with a user that can access the content overview.
+    $this->drupalLogout();
+    $this->drupalLogin($this->drupalCreateUser(['access toolbar', 'access content overview']));
+    $this->assertResponse(200);
+    // @todo The toolbar div itself still has the id "toolbar-administration".
+    // @see https://www.drupal.org/project/drupal/issues/1044090
+    $this->assertSession()->elementExists('css', 'div[id=toolbar-administration]');
+    $this->assertSession()->elementExists('css', 'a[id=toolbar-item-administration]');
+  }
+
+  /**
    * Get the hash value from the admin menu subtrees route path.
    *
    * @return string
diff --git a/core/modules/toolbar/toolbar.module b/core/modules/toolbar/toolbar.module
index 7df073f..fdb8617 100644
--- a/core/modules/toolbar/toolbar.module
+++ b/core/modules/toolbar/toolbar.module
@@ -159,6 +159,35 @@ function toolbar_toolbar() {
     '#weight' => -20,
   ];
 
+  // If the current user cannot access administration pages, we can save a large
+  // amount of unnecessary work by ending here. It'd be better to actually know
+  // if the admin menu tree is empty for them, but trying to load that tree only
+  // happens in a #pre_render callback, and at that point, it's too late. The
+  // entire toolbar is rendered with the 'user.permissions' #cache context, so
+  // we can safely do this here and it'll still be cached correctly.
+  // @see toolbar_prerender_toolbar_administration_tray()
+  // @see toolbar_page_top()
+  $current_user = \Drupal::currentUser();
+  $current_user_is_admin = FALSE;
+  // @todo Make this array configurable somehow?
+  $admin_permissions = [
+    'access administration pages',
+    'access content overview',
+    'access media overview',
+    'administer blocks',
+    'administer comments',
+    'administer nodes',
+  ];
+  foreach ($admin_permissions as $admin_permission) {
+    if ($current_user->hasPermission($admin_permission)) {
+      $current_user_is_admin = TRUE;
+      break;
+    }
+  }
+  if (!$current_user_is_admin) {
+    return $items;
+  }
+
   // To conserve bandwidth, we only include the top-level links in the HTML.
   // The subtrees are fetched through a JSONP script that is generated at the
   // toolbar_subtrees route. We provide the JavaScript requesting that JSONP
