diff --git a/core/includes/theme.inc b/core/includes/theme.inc
index fbf78cf66b3..6e051a6e060 100644
--- a/core/includes/theme.inc
+++ b/core/includes/theme.inc
@@ -535,6 +535,12 @@ function template_preprocess_datetime_form(&$variables) {
  */
 function template_preprocess_datetime_wrapper(&$variables) {
   $element = $variables['element'];
+  $element += [
+    '#title_display' => 'before',
+    '#wrapper_attributes' => [],
+    '#label_attributes' => [],
+  ];
+  $variables['attributes'] = $element['#wrapper_attributes'];
 
   if (!empty($element['#title'])) {
     $variables['title'] = $element['#title'];
@@ -544,6 +550,19 @@ function template_preprocess_datetime_wrapper(&$variables) {
       $variables['title'] = ['#markup' => $variables['title']];
     }
   }
+  // Add label_display and label variables to template.
+  $variables['label_display'] = $element['#title_display'];
+  $variables['label'] = ['#theme' => 'form_element_label'];
+  $variables['label'] += array_intersect_key($element, array_flip(['#id', '#required', '#title', '#title_display']));
+  $variables['label']['#attributes'] = $element['#label_attributes'];
+
+  // Pass elements #type and #name to template.
+  if (!empty($element['#type'])) {
+    $variables['type'] = $element['#type'];
+  }
+  if (!empty($element['#name'])) {
+    $variables['name'] = $element['#name'];
+  }
 
   // Suppress error messages.
   $variables['errors'] = NULL;
diff --git a/core/lib/Drupal/Core/Datetime/Element/Datetime.php b/core/lib/Drupal/Core/Datetime/Element/Datetime.php
index e877971bd92..a1d4400f9af 100644
--- a/core/lib/Drupal/Core/Datetime/Element/Datetime.php
+++ b/core/lib/Drupal/Core/Datetime/Element/Datetime.php
@@ -266,6 +266,7 @@ public static function processDatetime(&$element, FormStateInterface $form_state
         '#size' => max(12, strlen($element['#value']['date'])),
         '#error_no_message' => TRUE,
         '#date_date_format' => $element['#date_date_format'],
+        '#states' => empty($element['#states']) ? [] : $element['#states'],
       ];
 
       // Allows custom callbacks to alter the element.
@@ -296,6 +297,7 @@ public static function processDatetime(&$element, FormStateInterface $form_state
         '#required' => $element['#required'],
         '#size' => 12,
         '#error_no_message' => TRUE,
+        '#states' => empty($element['#states']) ? [] : $element['#states'],
       ];
 
       // Allows custom callbacks to alter the element.
diff --git a/core/misc/states.js b/core/misc/states.js
index 6db3494cd7b..a8cd1daf1c9 100644
--- a/core/misc/states.js
+++ b/core/misc/states.js
@@ -701,6 +701,9 @@
         .find(tagsSupportDisable)
         .addBack(tagsSupportDisable)
         .prop('disabled', e.value);
+      $(e.target)
+        .closest('.js-complex-form-item')
+        .toggleClass('form-disabled', e.value);
     }
   });
 
diff --git a/core/modules/system/templates/datetime-wrapper.html.twig b/core/modules/system/templates/datetime-wrapper.html.twig
index a14da8937bd..81f00604ffa 100644
--- a/core/modules/system/templates/datetime-wrapper.html.twig
+++ b/core/modules/system/templates/datetime-wrapper.html.twig
@@ -5,8 +5,16 @@
  *
  * Available variables:
  * - content: The form element to be output, usually a datelist, or datetime.
- * - title: The title of the form element.
- * - title_attributes: HTML attributes for the title wrapper.
+ * - label: A rendered label element.
+ * - label_display: Label display setting. It can have these values:
+ *   - before: The label is output before the element. This is the default.
+ *     The label includes the #title and the required marker, if #required.
+ *   - after: The label is output after the element. For example, this is used
+ *     for radio and checkbox #type elements. If the #title is empty but the
+ *     field is #required, the label will contain only the required marker.
+ *   - invisible: Labels are critical for screen readers to enable them to
+ *     properly navigate through forms but can be visually distracting. This
+ *     property hides the label for everyone except screen readers.
  * - description: Description text for the form element.
  * - required: An indicator for whether the associated form element is required.
  *
@@ -16,22 +24,26 @@
  */
 #}
 {%
-  set title_classes = [
-    required ? 'js-form-required',
-    required ? 'form-required',
-  ]
+set container_classes = [
+'js-complex-form-item',
+]
 %}
-{% if title %}
-  <h4{{ title_attributes.addClass(title_classes) }}>{{ title }}</h4>
-{% endif %}
-{{ content }}
-{% if errors %}
-  <div>
-    {{ errors }}
-  </div>
-{% endif %}
-{% if description %}
-  <div{{ description_attributes }}>
-    {{ description }}
-  </div>
-{% endif %}
+<div{{ attributes.addClass(container_classes) }}>
+  {% if label_display in ['before', 'invisible'] %}
+    {{ label }}
+  {% endif %}
+  {{ content }}
+  {% if errors %}
+    <div>
+      {{ errors }}
+    </div>
+  {% endif %}
+  {% if description %}
+    <div{{ description_attributes }}>
+      {{ description }}
+    </div>
+  {% endif %}
+  {% if label_display == 'after' %}
+    {{ label }}
+  {% endif %}
+</div>
diff --git a/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.info.yml b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.info.yml
new file mode 100644
index 00000000000..65d61142dcb
--- /dev/null
+++ b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.info.yml
@@ -0,0 +1,5 @@
+name: 'Datetime #states test'
+type: module
+description: 'Provides an example demonstrating how form #states affect datetime elements.'
+package: Testing
+version: VERSION
diff --git a/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.info.yml.~1~ b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.info.yml.~1~
new file mode 100644
index 00000000000..e69de29bb2d
diff --git a/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.routing.yml b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.routing.yml
new file mode 100644
index 00000000000..6594e48a8b2
--- /dev/null
+++ b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.routing.yml
@@ -0,0 +1,13 @@
+datetime_states_test.example:
+  path: '/datetime-states-test/example'
+  defaults:
+    _form: '\Drupal\datetime_states_test\Form\DatetimeStatesTestForm'
+  requirements:
+    _access: 'TRUE'
+
+datetime_states_test.admin.example:
+  path: '/admin/datetime-states-test/example'
+  defaults:
+    _form: '\Drupal\datetime_states_test\Form\DatetimeStatesTestForm'
+  requirements:
+    _access: 'TRUE'
diff --git a/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.routing.yml.~1~ b/core/modules/system/tests/modules/datetime_states_test/datetime_states_test.routing.yml.~1~
new file mode 100644
index 00000000000..e69de29bb2d
diff --git a/core/modules/system/tests/modules/datetime_states_test/src/Form/DatetimeStatesTestForm.php b/core/modules/system/tests/modules/datetime_states_test/src/Form/DatetimeStatesTestForm.php
new file mode 100644
index 00000000000..186711ce4a3
--- /dev/null
+++ b/core/modules/system/tests/modules/datetime_states_test/src/Form/DatetimeStatesTestForm.php
@@ -0,0 +1,72 @@
+<?php
+
+namespace Drupal\datetime_states_test\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Defines a form for testing #states on datetime elements.
+ *
+ * Includes toggle checkboxes for 'Invisible', 'Disabled' and 'Required', and a
+ * datetime field with #states configured appropriately for each checkbox.
+ */
+class DatetimeStatesTestForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'datetime_states_test_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['toggle_invisible'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Invisible'),
+    ];
+    $form['toggle_disabled'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Disabled'),
+    ];
+    $form['toggle_required'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Required'),
+    ];
+
+    $form['datetime'] = [
+      '#type' => 'datetime',
+      '#title' => $this->t('Datetime'),
+      '#description' => $this->t('A datetime form element.'),
+      '#states' => [
+        'invisible' => [
+          ':input[name="toggle_invisible"]' => ['checked' => TRUE],
+        ],
+        'disabled' => [
+          ':input[name="toggle_disabled"]' => ['checked' => TRUE],
+        ],
+        'required' => [
+          ':input[name="toggle_required"]' => ['checked' => TRUE],
+        ],
+      ],
+    ];
+
+    $form['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Save'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    // Sample form only, nothing to submit.
+  }
+
+}
diff --git a/core/modules/system/tests/modules/datetime_states_test/src/Form/DatetimeStatesTestForm.php.~1~ b/core/modules/system/tests/modules/datetime_states_test/src/Form/DatetimeStatesTestForm.php.~1~
new file mode 100644
index 00000000000..e69de29bb2d
diff --git a/core/modules/system/tests/src/FunctionalJavascript/DatetimeStatesTest.php b/core/modules/system/tests/src/FunctionalJavascript/DatetimeStatesTest.php
new file mode 100644
index 00000000000..e9b2ac0a95b
--- /dev/null
+++ b/core/modules/system/tests/src/FunctionalJavascript/DatetimeStatesTest.php
@@ -0,0 +1,115 @@
+<?php
+
+namespace Drupal\Tests\system\FunctionalJavascript;
+
+use Drupal\Core\Url;
+use Drupal\FunctionalJavascriptTests\WebDriverTestBase;
+
+/**
+ * Ensure that #states works properly on form elements of #type 'datetime'.
+ *
+ * @group datetime
+ */
+class DatetimeStatesTest extends WebDriverTestBase {
+
+  /**
+   * The user object to test with.
+   *
+   * @var \Drupal\user\UserInterface
+   */
+  protected $account;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = ['datetime_states_test'];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Create a user.
+    $account = $this->drupalCreateUser();
+
+    // Activate user by logging in.
+    $this->drupalLogin($account);
+  }
+
+  /**
+   * Tests #states functionality on datetime form elements.
+   */
+  public function testDatetimeStates() {
+    $this->drupalGet(Url::fromRoute('datetime_states_test.example'));
+
+    // Make sure the initial state makes sense.
+    $page = $this->getSession()->getPage();
+
+    $toggle_invisible = $page->findField('toggle_invisible');
+    $this->assertNotEmpty($toggle_invisible);
+    $this->assertTrue($toggle_invisible->isVisible());
+    $this->assertNoFieldChecked('edit-toggle-invisible');
+
+    $toggle_disabled = $page->findField('toggle_disabled');
+    $this->assertNotEmpty($toggle_disabled);
+    $this->assertTrue($toggle_disabled->isVisible());
+    $this->assertNoFieldChecked('edit-toggle-disabled');
+
+    $toggle_required = $page->findField('toggle_required');
+    $this->assertNotEmpty($toggle_required);
+    $this->assertTrue($toggle_required->isVisible());
+    $this->assertNoFieldChecked('edit-toggle-required');
+
+    $datetime_label = $this->assertSession()->elementExists('css', 'label[for="edit-datetime"]');
+    $this->assertNotEmpty($datetime_label);
+    $this->assertTrue($datetime_label->isVisible(), 'Datetime label is visible');
+    $this->assert($datetime_label->getText() === 'Datetime', 'Datetime label is correct');
+    $this->assertFalse($datetime_label->hasClass('js-form-required'), 'Datetime label is not marked as required.');
+
+    $date_field = $page->findField('datetime[date]');
+    $this->assertNotEmpty($date_field);
+    $this->assertTrue($date_field->isVisible(), 'Date is visible');
+    $this->assertFalse($date_field->hasAttribute('disabled'), 'Date is enabled');
+    $this->assertFalse($date_field->hasAttribute('required'), 'Date is optional');
+
+    $time_field = $page->findField('datetime[time]');
+    $this->assertNotEmpty($time_field);
+    $this->assertTrue($time_field->isVisible(), 'Time is visible');
+    $this->assertFalse($time_field->hasAttribute('disabled'), 'Time is enabled');
+    $this->assertFalse($time_field->hasAttribute('required'), 'Time is optional');
+
+    // Now, check the 'Invisible' toggle and see what happens.
+    $toggle_invisible->check();
+    $this->assertFalse($date_field->isVisible(), 'Date is invisible');
+    $this->assertFalse($time_field->isVisible(), 'Time is invisible');
+    $this->assertFalse($datetime_label->isVisible(), 'Datetime label is invisible');
+
+    // Make it visible again.
+    $toggle_invisible->uncheck();
+    $this->assertTrue($date_field->isVisible(), 'Date is visible');
+    $this->assertTrue($time_field->isVisible(), 'Time is visible');
+    $this->assertTrue($datetime_label->isVisible(), 'Datetime label is visible');
+
+    // Make it disabled.
+    $toggle_disabled->check();
+    $this->assertTrue($date_field->hasAttribute('disabled'), 'Date is disabled');
+    $this->assertTrue($time_field->hasAttribute('disabled'), 'Time is disabled');
+    // Put it back.
+    $toggle_disabled->uncheck();
+    $this->assertFalse($date_field->hasAttribute('disabled'), 'Date is enabled');
+    $this->assertFalse($time_field->hasAttribute('disabled'), 'Time is enabled');
+
+    // Make it required.
+    $toggle_required->check();
+    $this->assertTrue($date_field->hasAttribute('required'), 'Date is required');
+    $this->assertTrue($time_field->hasAttribute('required'), 'Time is required');
+    $this->assertTrue($datetime_label->hasClass('js-form-required'), 'Datetime label is marked as required');
+  }
+
+}
diff --git a/core/themes/claro/templates/datetime-wrapper.html.twig b/core/themes/claro/templates/datetime-wrapper.html.twig
index 229e8b60396..4d50759da10 100644
--- a/core/themes/claro/templates/datetime-wrapper.html.twig
+++ b/core/themes/claro/templates/datetime-wrapper.html.twig
@@ -6,17 +6,15 @@
  * @see template_preprocess_form_element()
  */
 #}
+{% if errors %}
 {%
-  set title_classes = [
-    'form-item__label',
-    required ? 'js-form-required',
-    required ? 'form-required',
-    errors ? 'has-error',
-  ]
+  set label_attributes = label_attributes.addClass('has-error')
 %}
-<div class="form-item form-datetime-wrapper">
-{% if title %}
-  <h4{{ title_attributes.addClass(title_classes) }}>{{ title }}</h4>
+{% endif %}
+
+<div class="form-item form-datetime-wrapper js-complex-form-item">
+{% if label_display in ['before', 'invisible'] %}
+  {{ label }}
 {% endif %}
 {{ content }}
 {% if errors %}
@@ -29,4 +27,7 @@
     {{ description }}
   </div>
 {% endif %}
+{% if label_display == 'after' %}
+  {{ label }}
+{% endif %}
 </div>
