diff --git a/core/modules/datetime/src/Plugin/views/filter/Date.php b/core/modules/datetime/src/Plugin/views/filter/Date.php
index 4211d59..a401cdb 100644
--- a/core/modules/datetime/src/Plugin/views/filter/Date.php
+++ b/core/modules/datetime/src/Plugin/views/filter/Date.php
@@ -201,6 +201,11 @@ public function buildExposedForm(&$form, FormStateInterface $form_state) {
     // @see \Drupal\views\Plugin\views\filter\NumericFilter::valueForm()
     if ($this->fieldStorageDefinition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE) {
       $field_identifier = $this->options['expose']['identifier'];
+      // The form elements might be encased in a wrapper, so check that first.
+      $field_wrapper = $field_identifier . '_wrapper';
+      if (isset($form[$field_wrapper])) {
+        $this->alterFormElementGranularity($form[$field_wrapper], 'day');
+      }
       if (isset($form[$field_identifier])) {
         $this->alterFormElementGranularity($form[$field_identifier], 'day');
       }
diff --git a/core/modules/views/src/Plugin/views/filter/Date.php b/core/modules/views/src/Plugin/views/filter/Date.php
index 7f57182..45a2414 100644
--- a/core/modules/views/src/Plugin/views/filter/Date.php
+++ b/core/modules/views/src/Plugin/views/filter/Date.php
@@ -232,7 +232,12 @@ public function buildExposedForm(&$form, FormStateInterface $form_state) {
       // for anything of type 'textfield', and convert it to 'datetime'.
       // @see \Drupal\views\Plugin\views\filter\NumericFilter::valueForm()
       $field_identifier = $this->options['expose']['identifier'];
-      if (isset($form[$field_identifier])) {
+      // The form elements might be encased in a wrapper, so check that first.
+      $field_wrapper = $field_identifier . '_wrapper';
+      if (isset($form[$field_wrapper])) {
+        $this->convertTextElementToDatetime($form[$field_wrapper]);
+      }
+      elseif (isset($form[$field_identifier])) {
         $this->convertTextElementToDatetime($form[$field_identifier]);
       }
 
