diff --git a/core/modules/toolbar/js/toolbar.es6.js b/core/modules/toolbar/js/toolbar.es6.js
index 017e91112a..226fcbb5cf 100644
--- a/core/modules/toolbar/js/toolbar.es6.js
+++ b/core/modules/toolbar/js/toolbar.es6.js
@@ -50,9 +50,7 @@
           locked: JSON.parse(
             localStorage.getItem('Drupal.toolbar.trayVerticalLocked'),
           ),
-          activeTab: document.getElementById(
-            JSON.parse(localStorage.getItem('Drupal.toolbar.activeTabID')),
-          ),
+          activeTab: null,
           height: $('#toolbar-administration').outerHeight(),
         });
 
@@ -96,7 +94,6 @@
         // Force layout render to fix mobile view. Only needed on load, not
         // for every media query match.
         model.trigger('change:isFixed', model, model.get('isFixed'));
-        model.trigger('change:activeTray', model, model.get('activeTray'));
 
         // Render collapsible menus.
         const menuModel = new Drupal.toolbar.MenuModel();
@@ -150,21 +147,6 @@
             $(document).trigger('drupalToolbarTrayChange', tray);
           });
 
-        // If the toolbar's orientation is horizontal and no active tab is
-        // defined then show the tray of the first toolbar tab by default (but
-        // not the first 'Home' toolbar tab).
-        if (
-          Drupal.toolbar.models.toolbarModel.get('orientation') ===
-            'horizontal' &&
-          Drupal.toolbar.models.toolbarModel.get('activeTab') === null
-        ) {
-          Drupal.toolbar.models.toolbarModel.set({
-            activeTab: $(
-              '.toolbar-bar .toolbar-tab:not(.home-toolbar-tab) a',
-            ).get(0),
-          });
-        }
-
         $(window).on({
           'dialog:aftercreate': (event, dialog, $element, settings) => {
             const $toolbar = $('#toolbar-bar');
diff --git a/core/modules/toolbar/js/toolbar.js b/core/modules/toolbar/js/toolbar.js
index 47177120d7..e2e0eab786 100644
--- a/core/modules/toolbar/js/toolbar.js
+++ b/core/modules/toolbar/js/toolbar.js
@@ -25,7 +25,7 @@
       once('toolbar', '#toolbar-administration', context).forEach(function (toolbar) {
         var model = new Drupal.toolbar.ToolbarModel({
           locked: JSON.parse(localStorage.getItem('Drupal.toolbar.trayVerticalLocked')),
-          activeTab: document.getElementById(JSON.parse(localStorage.getItem('Drupal.toolbar.activeTabID'))),
+          activeTab: null,
           height: $('#toolbar-administration').outerHeight()
         });
         Drupal.toolbar.models.toolbarModel = model;
@@ -51,7 +51,6 @@
           model: model
         });
         model.trigger('change:isFixed', model, model.get('isFixed'));
-        model.trigger('change:activeTray', model, model.get('activeTray'));
         var menuModel = new Drupal.toolbar.MenuModel();
         Drupal.toolbar.models.menuModel = menuModel;
         Drupal.toolbar.views.menuVisualView = new Drupal.toolbar.MenuVisualView({
@@ -76,11 +75,6 @@
         }).on('change:activeTray', function (model, tray) {
           $(document).trigger('drupalToolbarTrayChange', tray);
         });
-        if (Drupal.toolbar.models.toolbarModel.get('orientation') === 'horizontal' && Drupal.toolbar.models.toolbarModel.get('activeTab') === null) {
-          Drupal.toolbar.models.toolbarModel.set({
-            activeTab: $('.toolbar-bar .toolbar-tab:not(.home-toolbar-tab) a').get(0)
-          });
-        }
         $(window).on({
           'dialog:aftercreate': function dialogAftercreate(event, dialog, $element, settings) {
             var $toolbar = $('#toolbar-bar');
@@ -147,4 +141,4 @@
   Drupal.AjaxCommands.prototype.setToolbarSubtrees = function (ajax, response, status) {
     Drupal.toolbar.setSubtrees.resolve(response.subtrees);
   };
-})(jQuery, Drupal, drupalSettings);
\ No newline at end of file
+})(jQuery, Drupal, drupalSettings);
diff --git a/core/modules/toolbar/toolbar.module b/core/modules/toolbar/toolbar.module
index e02fa585ec..7ba4c1c3e0 100644
--- a/core/modules/toolbar/toolbar.module
+++ b/core/modules/toolbar/toolbar.module
@@ -252,7 +252,6 @@ function toolbar_preprocess_html(&$variables) {
   if (!\Drupal::currentUser()->hasPermission('access toolbar')) {
     return;
   }
-  $variables['attributes']['class'][] = 'toolbar-tray-open';
   $variables['attributes']['class'][] = 'toolbar-horizontal';
   $variables['attributes']['class'][] = 'toolbar-fixed';
   $variables['attributes']['class'][] = 'toolbar-loading';
